client/util.go

```diff
@@ -21,17 +22,15 @@ const docIDPrefixLength = docIDVersionLength + docIDNonceLength
 // pathnameToDocID encrypts a `pathname` to a document ID using `key`.
 func pathnameToDocID(pathname string, key [32]byte) (sserver1.DocumentID, error) {
 	var nonce [docIDNonceLength]byte
-	_, err := rand.Read(nonce[:])
-	if err != nil {
-		return sserver1.DocumentID(""), err
-	}
+	cksum := sha256.Sum256([]byte(pathname))
```

Comment:
This is not better, but the issue is that the encryption should be deterministic. Otherwise calculating the document ID using the same pathname would yield different results. This is a compromise that we have to make. But this is still secure, as the (key, nonce) pair is almost impossible to be reused.